Win11Debloat.ps1

```
@@ -220,6 +220,17 @@ else {
 }
 
+# Check if the device is domain-joined and warn the user (Group Policy may override changes)
+try {
+    $computerSystem = Get-CimInstance Win32_ComputerSystem -ErrorAction SilentlyContinue
+    if ($null -ne $computerSystem -and $computerSystem.PartOfDomain) {
+        Write-Warning "This machine is domain-joined. Group Policy may override changes made by Win11Debloat."
+    }
+}
+catch {
+    # Suppress WMI/CIM query and null reference errors to ensure the script does not crash on startup
+}
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)